values_test.go

```diff
@@ -635,6 +635,8 @@ func TestGetType(t *testing.T) {
 			reflect.TypeOf(Event{}):      {},
 			reflect.TypeOf(Resource{}):   {},
 			reflect.TypeOf(Struct{}):     {},
+			reflect.TypeOf(Enum{}):       {},
+			reflect.TypeOf(Attachment{}): {},
```

Comment:
So while extending these tests I noticed that the existing code relying on the `reflect.typelinks` function doesn't include these two types for whatever reason. I saw the code was taken from https://stackoverflow.com/questions/32132064/how-to-discover-all-package-types-at-runtime.

What is so special about `Enum` that it isn't included, but e.g. `Struct` is? 🤔

Reply:
Maybe because this file doesn't instantiate any `Enum` instances? (should have been a test case there for enums!)

Reply:
oh well, that looks like a problem with this test. It only can find types that are actually used inside this file 🥴

Reply:
So I tried to look a bit more into it and finally found and confirmed the problem:

`typenames` says:

```
// Only types we might want to look up are included:
// pointers, channels, maps, slices, and arrays.
```

If I add a simple statement like this, which converts to `any`, i.e. creates a pointer, `Enum` gets included:

```golang
fmt.Printf("%v\n", Enum{})
```

I can't find any alternative, so for now I'll remove the code relying on internals, and just manually list the `Value` implementations. This won't let us automatically handle new cases, but the current code may not even handle existing cases.